roles/matrix-synapse/vars/workers.yml

```diff
@@ -0,0 +1,308 @@
+---
+
+matrix_synapse_workers_generic_worker_endpoints:
```

Comment:
I'd argue that splitting these endpoints directly into `matrix_synapse_workers_generic_worker_endpoints.[sync|federation|...]`, and combining groups where needed would be more maintenance friendly.
It makes it easy to route only federation/only sync/... requests

Reply:
You're suggesting that we pre-split these early on, as part of the awk script?

I guess that would be better compared to the runtime splitting I did in 5ca68210cdd. But then again, one needs to rework the awk script somewhat, which.. I haven't dipped my toes into yet.

Reply:
I would suggest splitting into arrays of endpoints (`sync: [], federation: [], media: [], ...`), if it was generated with the awk script, it should also be added there, but having them split in the source allows other contributors to cleanly use f.ex. the set of all federation endpoints.